app/src/components/market/sections/market_buy/market_buy.tsx

```diff
@@ -499,10 +498,11 @@ const MarketBuyWrapper: React.FC<Props> = (props: Props) => {
       )}
       {showUpgrade && (
         <SetAllowance
-          collateral={nativeAsset}
+          collateral={isNative ? displayCollateral : undefined}
```

Comment:
Wouldn't we want this to always be undefined if this is just for upgrading the proxy and not setting allowance?

Reply:
True, I made this change to display our older messaging just for native markets but should remove to simplify.